mondrian/src/main/java/mondrian/olap/fun/FunUtil.java

```diff
@@ -2734,7 +2734,7 @@ static boolean existsInTuple(
             Member rightMember = getCorrespondingMember(
                 leftMember, rightTuple, rightHierarchies, eval);
             checkedMembers.add(rightMember);
-            if (!isOnSameHierarchyChain(leftMember, rightMember)) {
+           if (!(leftMember.isOnSameHierarchyChain(rightMember)||rightMember.isOnSameHierarchyChain(leftMember))) {
```

Comment:
This change is going to introduce a performance regression with the regular EXISTS function.  Suppose you were checking whether [Time].[2003] exists in the same hierarchy as [Time].[2004].[QTR1], before FunUtil.isAncestorOf would have been called twice.  After this change, it'll be called four times.  

Maybe it would have been cleaner to modify isOnSameHierarchyChain to check the type of the member and not try to add the isOnSameHierarchyChain to the Member interface.